docs/intro_tutorial.rst

```diff
@@ -22,14 +22,84 @@ do not, please consult the
+
+Quantum states :math:`\rho` characterize a particular way in which the system *entangles* the possible configurations of the register. One of the main reasons for defining quantum states like this is to have a straightforward way of expressing **mixtures** of pure states.
+
+A pure state, in terms of density matrices, is a quantum state that is built from the outer product of the unit vector with itself: :math:`\rho_{\text{pure}} = u* u \quad,` for :math:`u`  a unit vector. Mixed states generalize quantum states to conditions where we have only probabilistic knowledge of the member states. Moreover, a set of positive semi
+
+In general, mixed states will be **convex combinations** of pure quantum states. For larger systems, one may want to quickly check one has a mixture, so let's use :code:`toqito` , which verifies if the states are pure and construct their mixture:
+
+.. code-block:: python
+
+  L = 3 #System size
+  N = 6 #Ensemble size
+
+  # Produce some probabilities that add up to one.
+  prob = np.random.rand(N)
+  prob = prob/np.sum(prob)
+
+  print(f' Ensemble has probabilities {list(prob)} \n')
+
+  # Define randomly some pure states:
+  np.random.seed(1)
+  states = np.zeros((N,L,L))
+  for j in range (N):
+      # Build unit vector and calculate outer product
+      vector = np.random.randint(1,4,L)
+      matrix = np.outer(vector, vector)
+
+      states[j]=(matrix/np.trace(matrix))
+
+  print(f'Boolean check of density matrix: \n {[tqt.matrix_props.is_density(states[j]) for j in range(N)]} \n')
+  print(f'Our states are: \n {states} \n\n Boolean check of purity: \n {[tqt.state_props.is_pure(states[j]) for j in range(N)]} \n')
+  print(f'Do we have an ensemble? {tqt.state_props.is_ensemble([prob[i]*states[i] for i in range(N)])}\n')
+
+  # Produce convex sum:
+  convex_sum = np.zeros((L,L))
+  for i in range(N):
+      convex_sum += prob[i]*states[i]
+
+  print(f'The mixture of the states is: \n {convex_sum} \n')
+  print(f'Is the final state mixed? {tqt.state_props.is_mixed(convex_sum)}')
+
+
+We can also build common states quickly from :code:`toqito` . For example
+
 
 Channels
 --------
 
-(Coming soon).
+Channels are representations of *discrete changes* in a register. They are linear maps :math:`\Phi` from one space of square operators to another, such that :math:`\Phi(P)` is still a positive semidefinite operator and :math:`\mathrm{Tr} (\Phi(P)) = 1` (trace is preserved):
 
 Measurements
 ------------
 
-(Coming soon).
\ No newline at end of file
+When we extract (classical) information from a quantum system, outcomes are generally generated at random, and they depend the quantum state of the system at the moment of extracting information and on the measurement itself.
+
+More precisely, a **measurement** is a map :math:`\mu` from the different outcomes of our register (called an **alphabet** :math:`\Sigma`) to semidefinite matrices, such that the different measurement operators add up to a unit matrix:
```

Comment:
semidefinite should be positive semidefinite.

Reply:
Fixed